In [1]:
import pandas as pd

df = pd.read_csv('../data/diabetes.csv')

print('Shape:', df.shape)
print('Columns:', df.columns.tolist())

Shape: (768, 9)
Columns: ['Pregnancies', 'Glucose', 'BloodPressure', 'SkinThickness', 'Insulin', 'BMI', 'DiabetesPedigreeFunction', 'Age', 'Outcome']


In [2]:
print(df.dtypes)

Pregnancies                   int64
Glucose                       int64
BloodPressure                 int64
SkinThickness                 int64
Insulin                       int64
BMI                         float64
DiabetesPedigreeFunction    float64
Age                           int64
Outcome                       int64
dtype: object


In [3]:
print(df.describe())

       Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  

In [4]:
print(df['Outcome'].value_counts())

Outcome
0    500
1    268
Name: count, dtype: int64


In [5]:
diabetic = df[df['Outcome'] == 1]
non_diabetic = df[df['Outcome'] == 0]

print(f"Diabetic patients: {len(diabetic)}")
print(f"Non-Diabetic patients: {len(non_diabetic)}")
print(f"Mean glucose - Diabetic: {diabetic['Glucose'].mean():.1f}")
print(f"Mean glucose - Non-Diabetic: {non_diabetic['Glucose'].mean():.1f}")

Diabetic patients: 268
Non-Diabetic patients: 500
Mean glucose - Diabetic: 141.3
Mean glucose - Non-Diabetic: 110.0


In [6]:
print(f"Missing values:\n{df.isnull().sum()}")

Missing values:
Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64


## My Observations — Pima Diabetes Dataset

1. How many diabetic vs non-diabetic patients are in this dataset: 268/500

2. The average glucose level across all patients is: 120.89.

3. The average glucose level is higher in diabetic patients specifically: 141.3.

4. One thing that surprised me about this real data: The size, the outcomes of diabetes and hypertension in pregnant females.

5. Why I think the zeros in Glucose/BloodPressure/BMI are probably missing data rather than real measurements: It is impossible to have zero of all these values, it wasn't filled so the system filled it in as zero. I was told I will learn something about this in data cleaning.

## Day 16 Complete — First Real Dataset

This is the first time I have worked with actual patient data rather than invented examples. 

Confidence working with a real CSV file (honest, 1-10): 5